### ISSUE TITLE
cAdvisor starting with the error: inotify_init: too many open files

### ISSUE THREAD

**Issue by itechbird** [google/cadvisor#2363]
When I start cAdvisor (v0.33.0) in docker, the startup fails and the log prompts: "manager.go:341] Registration of the raw container factory failed: inotify_init: too many open files. cadvisor.go:180] Failed to start container manager: inotify_init: too many open files". 
But 'openfiles' has been configured to 600,000 and it still fails to start until I adjust the 'openfiles' to 1 million. 
I don't know how many 'openfiles' it needs, or does he mistakenly think that the file handle is abnormal or I need adjust  the other system parameter?
Thanks for answer.

**Comment by dashpole** [comment]
cAdvisor uses inotify to "watch" the entire cgroup tree for changes.  That is how it detects container creation and deletion.  As you might expect, doing this requires maintaining a large number of file descriptors.  The most likely answer to your question is "Just increase the limit", as it is expected that cAdvisor will use a large number of file descriptors.

One other avenue to check is how many open files there are before starting cAdvisor.  If it is close to your limit before it is started, that could also be the reason why cAdvisor hits the limit when starting up.

**Comment by youngboyvip** [comment]
Has anyone solved it ?

**Comment by youngboyvip** [comment]
Solved by
Increase inotify watch limits to Kubernetes instances.

max_user_instances = 8192,
max_user_watches = 524288,


https://github.com/jetstack/tarmak/pull/757

**Comment by prakashjha08** [comment]
On some EC2 Instances, CAdvisor works without any issues, without changing max_user_instances, max_user_watches. But on some, I get this error, which raises the question, do I really need to change max user watches and instances on the instances where CAdvisor is not working?

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
